include/seastar/core/condition-variable.hh

```diff
@@ -76,6 +76,8 @@ private:
     // the base for queue waiters. looks complicated, but this is
     // to make it transparent once we add non-promise based nodes
     struct waiter : public boost::intrusive::list_base_hook<boost::intrusive::link_mode<boost::intrusive::auto_unlink>> {
+        waiter() = default;
+        waiter(const waiter&) = default;
```

Comment:
It seems to me that waiters are not supposed to be copyable at all. Maybe the copy constructor should be explicitly deleted instead?

Reply:
indeed, i am adding the defaulted move constructor. it is what we need.